src/coprocessor/xeval/builtin_cast.rs

```diff
@@ -0,0 +1,392 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use tipb::expression::Expr;
+use coprocessor::codec::datum::Datum;
+use super::{Evaluator, EvalContext, Result, Error, ERROR_UNIMPLEMENTED};
+
+const TYPE_DURATION: &'static str = "duration";
+
+fn invalid_type_error(datum: &Datum, expected_type: &str) -> Result<Datum> {
+    Err(Error::Eval(format!("invalid expr type: {:?}, expect: {}", datum, expected_type)))
+}
+
+impl Evaluator {
+    pub fn cast_int_as_int(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
```

Comment:
Could you just implement castTimeAs* in this PR? There is too much other functions here, and it may confuse the reviewers.